metric/system/process/process_windows_test.go

```diff
@@ -0,0 +1,60 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package process
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent-system-metrics/metric/system/resolve"
+)
+
+func TestGetInfoForPid_numThreads(t *testing.T) {
+	const want = 42
+	// On windows the programs always start with more threads than the code
+	// is launching. It seems to be due to it loading dll/tables in parallel
+	// or de default thread pool. See the following for more info:
+	// https://stackoverflow.com/questions/42789199/why-there-are-three-unexpected-worker-threads-when-a-win32-console-application-s/42789684#42789684
+	expected := 45
```

Comment:
Can you ensure it's always 45 threads?

One option is to run the program for a while and use the eventually assertion to wait for when the number of threads goes back to 42

Reply:
it takes around 30s on my machine to have the number go back to 42. As it's been pretty deterministic, on my machine and on CI, I'm doing this double test. It accounts exactly for the extra threads, if something changes in the future the test sill fails and there should be enough explanation in the test for anyone else o understand the problem and fix it